judging/scripts/prescribe/generate_prescriptions.sh

```diff
@@ -27,13 +27,28 @@ generate_prescriptions_wrapper="$repo_dir/judging/scripts/prescribe/generate_pre
 prescription_module="$HOME/work/prescribe.py"
 validation_module="$repo_dir/judging/scripts/prescribe/prescriptor_validation.py"
 
-# Run script
+# Print out some environment diagnostics
 pwd
 command -v python
 command -v pip
 python --version
 pip --version
-python "$generate_prescriptions_wrapper" \
-  --requested-prescriptions-file "$prescriptions_file" \
-  --prescription-module "$prescription_module" \
-  --validation-module "$validation_module"
+
+# Temporarily disable errexit as we want to detect when flock fails so we can log a message
+set +o errexit
+
+# Run script within flock to prevent multiple instances if jobs overrun
+flock --nonblock /tmp/robojudge.lock \
+  python "$generate_prescriptions_wrapper" \
+    --requested-prescriptions-file "$prescriptions_file" \
+    --prescription-module "$prescription_module" \
+    --validation-module "$validation_module"
+retVal=$?
+
+if [ $retVal -ne 0 ]; then
+    echo "Unable to acquire lock. Previous job still running?"
+    echo "Processes:"
+    ps afx
```

Comment:
Thought this would help with troubleshooting and since it's in Docker the output is short.

Reply:
Don't we also get a non-zero `retVal` if the job fails, because `prescribe.py` doesn't exist for instance, or because it crashed?

Reply:
I wasn't sure so I tested this empirically. You are right -- if the underlying command fails, `flock` will return non-zero. And you can't even check the actual value of the `flock` return to see if it's "unable to obtain lock" as other scenarios generate the same value:

`flock /tmp/a touch /b`

yields rc == 1.

But worst case, it will result in a confusing error message in the log.